### PR TITLE
Migrate to github.com/TheThingsNetwork/api

### LINKED ISSUE #662: Migrate to github.com/TheThingsNetwork/api
We currently use `github.com/TheThingsNetwork/ttn/api`, but this is being moved to `github.com/TheThingsNetwork/api`.

In the process we are making some changes to the generated Go code in the new `api` package:

- `Id` is renamed to `ID`
- `Eui` is renamed to `EUI`
- Some fields are now marked as non-nullable

Please report any other improvements in https://github.com/TheThingsNetwork/api/issues

### PR DESCRIPTION
**Pull request by rvolosatovs**
Closes #662

### CODE REVIEW

**Comment by htdvisser** [comment]
```
ttnctl/cmd/components_check.go:33: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/devices_delete.go:32: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/devices_info.go:46: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/devices_personalize.go:36: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/devices_register.go:38: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/devices_register_on_join.go:33: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/devices_set.go:32: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/devices_simulate.go:23: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/downlink.go:37: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/gateways_delete.go:25: undefined: "github.com/TheThingsNetwork/ttn/api".NotEmptyAndValidID
ttnctl/cmd/gateways_delete.go:25: too many errors
```

**Comment by rvolosatovs** [comment]
I assumed CI tests `ttnctl` build also, will fix

**Comment by rvolosatovs** [comment]
Fixed the issue and rebased, should be ready to merge

File: core/band/adr.go [hunk 1]
```diff
@@ -39,15 +39,15 @@ func (f *FrequencyPlan) ADRSettings(dataRate string, txPower int, snr float32, d
 		return dataRate, txPower, ErrADRUnavailable
 	}
 	margin := linkMargin(dataRate, snr) - deviceMargin
-	drIdx, err := f.GetDataRateIndexFor(dataRate)
```
Comment (htdvisser) [inline core/band/adr.go]: Unnecessary, also below

File: core/band/band.go [hunk 2]
```diff
@@ -18,8 +18,8 @@ type FrequencyPlan struct {
 	CFList *lorawan.CFList
 }
 
-func (f *FrequencyPlan) GetDataRateStringForIndex(drIdx int) (string, error) {
```
Comment (htdvisser) [inline core/band/band.go]: Unnecessary

File: core/band/band_test.go [hunk 3]
```diff
@@ -114,12 +114,12 @@ func TestGetDataRate(t *testing.T) {
 
 	eu, _ := Get("EU_863_870")
 	euRates := []string{"SF12BW125", "SF11BW125", "SF10BW125", "SF9BW125", "SF8BW125", "SF7BW125", "SF7BW250"}
-	for expIdx, expRate := range euRates {
```
Comment (htdvisser) [inline core/band/band_test.go]: Unnecessary

**Pull request merged**